internal-api/src/main/java/datadog/trace/api/remoteconfig/ServiceNameCollector.java

```diff
@@ -48,14 +53,30 @@ public void addService(final String serviceName) {
       }
       return;
     }
-    if (!Config.get().getServiceName().equalsIgnoreCase(serviceName)) {
-      services.put(serviceName.toLowerCase(Locale.ROOT), serviceName);
-    }
+    services.put(serviceName, serviceName);
```

Comment:
```suggestion
    services.putIfAbsent(serviceName, serviceName);
```
this avoids any write operation when the key already exists; on later JDKs it also avoids taking an additional lock